### ISSUE TITLE
Unable to disable acknowledgement_timeout and auto_resolve_timeout

### ISSUE THREAD

**Issue by tigris** [terraform-providers/terraform-provider-pagerduty#51]
### Terraform Version

`Terraform v0.10.8`

### Affected Resource(s)

pagerduty_service

### Terraform Configuration Files

```hcl
resource "pagerduty_service" "dan_test" {
  name = "Dan Test Service"
  escalation_policy = "P5SFKTX"
}
```

### Debug Output

```
$ terraform apply
pagerduty_service.dan_test: Refreshing state... (ID: P8PH0G9)
pagerduty_service.dan_test: Modifying... (ID: P8PH0G9)
  acknowledgement_timeout: "1800" => "0"
  auto_resolve_timeout:    "14400" => "0"

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

### Panic Output

n/a

### Expected Behavior

Not changing the config, and doing an `apply` shouldn't modify the service, and the service should be created with disabled acknowledgement_timeout and auto_resolve_timeout.

### Actual Behavior

According to the UI, auto_resolve_timeout is set to 4 hours, and acknowledgement_timeout is set to 30 minutes. They're not "disabled".

Additionally, terraform will think there is a need to "reset" these values to `0` when doing a `plan` or an `apply`.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

I attempted to force these values to `null`, `nil` and `0`, but nothing has the desired effect of disabling these 2 values.

### References

n/a

**Comment by tigris** [comment]
Checking the pagerduty API https://v2.developer.pagerduty.com/v2/page/api-reference#!/Services/post_services it seems to disable these 2 features, you need to explicitly post `null` to the API. However, the validator for the config is not accepting `null` as a valid value for those fields, and if you omit those fields, it is posting a `0` (instead of `null`).

I believe this might be a simple fix, but I'm not that familiar with golang, I will see if I have some time later this week.

**Comment by pdecat** [comment]
Related to #44 which was merged but not released yet.

However, I've found there is still an issue as using 0 actually makes them 0s timeout rather than disabling them: incidents are resolved instantaneously.

Working on that right now.

**Comment by pdecat** [comment]
@tigris, I'm coming to same conclusion as you, the type of those fields is incorrect as it does not allows for passing null values.

I've got a fix, preparing a PR...

**Comment by tigris** [comment]
> Also, when the timeouts are set to 0 via the API, the web UI shows them as disabled on the view settings page[1], while the edit settings page shows them as checked with the minimal possible value of 10minutes selected[2]

Hmm, I'm seeing different behaviour. Actually when you set to 0 via the API, it is essentially a no-op. As in the settings will remain whatever they were previously, but no errors are returned.

**Comment by pdecat** [comment]
@tigris, this was the case without my previous PR #44 which was merged but not yet released.

**Comment by pdecat** [comment]
Hi @heimweh, #52 is also required to actually close this issue. Thanks!

**Comment by dustyketchum** [comment]
please reopen and merge #52 :)

**Comment by igorinsky** [comment]
Seems like the issue is still active.

Passing null to auto_resolve_timeout leads to updating the value to "14400". 
Also tried to remove auto_resolve_timeout from resource: plan says it will change the value to "14400" (and it really changes it this way, I tried to apply for one service).

So I've disabled this value manually in UI and terraform still suggests to change it to 14400.

I guess we will have to disable all the required services manually and to add an ignore_changes block.
